### PR TITLE
Refactor 3 catacurses functions a bit

### PR DESCRIPTION
**Pull request by AMurkin**
#### Summary
`SUMMARY: None`

#### Purpose of change
To reduce line count and make the code easier to read.

#### Describe the solution
Add temp variables. +3 minor edits.

#### Testing
It compiles and runs with no visual difference.

### CODE REVIEW

File: src/cursesport.cpp [hunk 1]
```diff
@@ -129,103 +128,54 @@ void catacurses::wborder( const window &win_, chtype ls, chtype rs, chtype ts, c
...
     }
-
-    if( bl ) {
-        mvwaddch( win_, point( 0, win->height - 1 ), bl );
-    } else {
-        mvwaddch( win_, point( 0, win->height - 1 ), LINE_XXOO );
+    for( i = 1; i < win->width - 1; i++ ) {
+        mvwaddch( win_, point( i, win->height - 1 ), border_bs );
     }
+    mvwaddch( win_, point_zero, tl ? tl : LINE_OXXO );
```
Comment (ifreund) [inline src/cursesport.cpp]: you factored out every other border `chtype` into a constant, why not this one as well for consistency?

File: src/cursesport.cpp [hunk 2]
```diff
@@ -129,103 +128,54 @@ void catacurses::wborder( const window &win_, chtype ls, chtype rs, chtype ts, c
...
-    } else {
-        for( i = 1; i < win->width - 1; i++ ) {
-            mvwaddch( win_, point( i, 0 ), LINE_OXOX );
-        }
-    }
+    const chtype border_ls = ls ? ls : LINE_XOXO;
+    const chtype border_rs = rs ? rs : LINE_XOXO;
+    const chtype border_ts = ts ? ts : LINE_OXOX;
+    const chtype border_bs = bs ? bs : LINE_OXOX;
+    const chtype border_tr = tr ? tr : LINE_OOXX;
```
Comment (ZhilkinSerg) [inline src/cursesport.cpp]:
```suggestion
    const chtype border_tl = tl ? tl : LINE_XOXO;
    const chtype border_tr = tr ? tr : LINE_OOXX;
```

File: src/cursesport.cpp (hunk 1 above)
Comment (ZhilkinSerg) [inline src/cursesport.cpp]:
```suggestion
    mvwaddch( win_, point_zero, border_tl );
```

Comment (AMurkin) [reply src/cursesport.cpp]: Missed one. Thanks.

**Pull request merged**